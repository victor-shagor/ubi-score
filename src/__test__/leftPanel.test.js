import LeftPanel from "components/auth/leftPanel";
import { mount, shallow } from "enzyme";
import toJson from "enzyme-to-json";

const mockProp = { name: "Name", role: "Role", text: "text" };

it("renders without crashing", () => {
  shallow(<LeftPanel />);
});

it("should render span", () => {
  const wrapper = mount(<LeftPanel name="Name" role="role" />);
  const value = wrapper.find("span").text();
  expect(value).toEqual("role");
});

it("should render two paragraphs", () => {
  const wrapper = mount(<LeftPanel {...mockProp} />);
  expect(wrapper.find("p").length).toEqual(2);
});

it("should render header", () => {
  const wrapper = shallow(<LeftPanel {...mockProp} />);
  expect(wrapper.contains(<p className="makeStyles-name-4">Name</p>)).toEqual(
    true
  );
});

it("should accept props", () => {
  const wrapper = mount(<LeftPanel name="Name" />);
  expect(wrapper.props().name).toEqual("Name");
});

it("renders correctly", () => {
  const tree = shallow(<LeftPanel {...mockProp} />);
  expect(toJson(tree)).toMatchSnapshot();
});
