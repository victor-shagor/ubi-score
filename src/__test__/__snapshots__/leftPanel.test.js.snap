// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="makeStyles-wrapper-1"
>
  <img
    className="makeStyles-circle-2"
    src="circle.png"
  />
  <img
    className="makeStyles-triangle-3"
    src="triangle.png"
  />
  <p
    className="makeStyles-name-4"
  >
    Name
  </p>
  <span
    className="makeStyles-role-5"
  >
    Role
  </span>
  <p
    className="makeStyles-text-6"
  >
    text
  </p>
  <Styled(MuiBox)
    display="flex"
  >
    <Styled(MuiBox)
      bgcolor="white"
      borderRadius="10px"
      height="4px"
      width="40px"
    />
    <Styled(MuiBox)
      bgcolor="white"
      borderRadius="10px"
      className="makeStyles-box-7"
      height="5px"
      marginLeft="7px"
      width="20px"
    />
    <Styled(MuiBox)
      bgcolor="white"
      borderRadius="10px"
      className="makeStyles-box-7"
      height="5px"
      marginLeft="7px"
      width="20px"
    />
  </Styled(MuiBox)>
</div>
`;
